src/k2/cpo/service/CPOService.cpp

```diff
@@ -268,9 +458,36 @@ std::tuple<Status, dto::Collection> CPOService::_getCollection(String name) {
     };
     K2INFO("Found collection in: " << cpath);
     std::get<0>(result) = Statuses::S200_OK("collection found");
+
+    // Check to see if we need to load schemas from file too
+    auto it = schemas.find(name);
+    if (it == schemas.end()) {
+        Status schemaStatus = _loadSchemas(name);
+        if (!schemaStatus.is2xxOK()) {
+            std::get<0>(result) = Statuses::S500_Internal_Server_Error("unable to read schema data");
+            return result;
+        }
+    }
+
     return result;
 }
 
+Status CPOService::_loadSchemas(const String& collectionName) {
+    auto cpath = _getCollectionPath(collectionName) + ".schemas";
```

Comment:
i see the ".schemas" append in two places. You should extract that into a helper method _getSchemasPath(collName)